pom.xml

```diff
@@ -592,10 +592,39 @@
                     <autoReleaseAfterClose>false</autoReleaseAfterClose>
                 </configuration>
             </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-clean-plugin</artifactId>
+                <version>3.2.0</version>
+                <executions>
+                    <execution>
+                        <!-- Rebuilding the manifest with the maven-bundle-plugin produces different results
+                             https://issues.apache.org/jira/browse/FELIX-6603
+                             Deleting the previously generated manifest is a temporary workaround.
+                             -->
+                        <id>bundle-manifest-fix</id>
+                        <phase>process-resources</phase>
+                        <goals>
+                            <goal>clean</goal>
+                        </goals>
+                        <configuration>
+                            <excludeDefaultDirectories>true</excludeDefaultDirectories>
+                            <filesets>
+                                <fileset>
+                                    <directory>${project.build.outputDirectory}</directory>
+                                    <includes>
+                                        <include>META-INF/MANIFEST.MF</include>
+                                    </includes>
+                                </fileset>
+                            </filesets>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
             <plugin>
                 <groupId>org.apache.felix</groupId>
                 <artifactId>maven-bundle-plugin</artifactId>
-                <version>3.3.0</version>
+                <version>5.1.8</version>
```

Comment:
IIRC, this will fail CI on the older JVMs
PR is marked as draft as we likely want to wait a bit on this one

Reply:
I know this is probably counter to reproducibility, but you can set different versions for different JVMs in the `<profiles>` section at the bottom as a version property.  Or have a different plugin configuration entirely.  Just a thought.

Reply:
That could work 🤔 
I've been wanting to try out a different idea too, building with a newer JDK, and using the toolchain option in the Surefire plugin to test with 7.
https://github.com/actions/setup-java/#maven-options

But, I have a feeling that's more work than it's worth, given that support for 7 will be dropped. 🤷

Reply:
Yeah, I dunno how much work you'd want to put in for JDK 7.  We won't need it much longer.

Reply:
I updated using the profile technique used, it's not worth spending more time with 7 😄